cpp/src/arrow/array.h

```diff
@@ -650,6 +650,36 @@ class ARROW_EXPORT FixedSizeBinaryArray : public PrimitiveArray {
   int32_t byte_width_;
 };
 
+/// DayTimeArray
+/// ---------------------
+/// \brief Array of Day and Millisecond values.
```

Comment:
FWIW, I'm a bit disappointed that we aren't deprecating this type =/

Reply:
I didn't want to get dragged into what it means deprecate this, should be easy enough to do if we want it.  As noted below I tried to say it was optional which I seem to recall seeing this was ok on a previous ML thread.